File: frontend/src/types/session.ts
export interface SessionMessage {
  id: string
  session_id: string
  role: 'user' | 'assistant'
  content: string
  sources?: string
  web_search_results?: string  // JSON 格式的网络搜索结果
  created_at: string
}

export interface Session {
  id: string
  collection_id: string
  title: string | null
  summary: string | null
  message_count: number
  is_active: boolean
  web_search_enabled: boolean
  created_at: string
  updated_at: string
  messages?: SessionMessage[]
}

export interface SessionListResponse {
  sessions: Session[]
  total: number
}

File: backend/app/schemas/session.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class SessionMessageBase(BaseModel):
    role: str
    content: str


class SessionMessageResponse(SessionMessageBase):
    id: str
    session_id: str
    sources: Optional[str] = None
    web_search_results: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True


class SessionCreate(BaseModel):
    collection_id: str
    title: Optional[str] = None
    web_search_enabled: bool = False


class SessionUpdate(BaseModel):
    title: Optional[str] = None
    web_search_enabled: Optional[bool] = None


class SessionResponse(BaseModel):
    id: str
    collection_id: str
    title: Optional[str]
    summary: Optional[str]
    message_count: int
    is_active: bool
    web_search_enabled: bool = False
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class SessionWithMessages(SessionResponse):
    messages: List[SessionMessageResponse] = []


class SessionListResponse(BaseModel):
    sessions: List[SessionResponse]
    total: int
